frontend/src/concepts/pipelines/content/modelCustomizationForm/modelCustomizationFormSchema/hyperparameterValidationUtils.ts

```diff
@@ -0,0 +1,57 @@
+import { z } from 'zod';
+import { InputDefinitionParameterType, ParametersKF } from '~/concepts/pipelines/kfTypes';
+
+const stringSchema = z.string().min(1, 'Hyperparameter length must be greater than 0');
+
+const getZodTypeFromParameterType = (paramType: InputDefinitionParameterType) => {
+  const structSchema = z
+    .string()
+    .trim()
+    .min(1, { message: 'Valid struct length has to be greater than 0' })
+    .refine(
+      (value) => {
+        try {
+          JSON.parse(value);
+          return true;
+        } catch (e) {
+          return false;
+        }
+      },
+      { message: 'Struct is not valid JSON' },
+    );
+  switch (paramType) {
+    case InputDefinitionParameterType.DOUBLE:
+      return z.number().positive();
+    case InputDefinitionParameterType.INTEGER:
+      return z.number().int().positive();
+    case InputDefinitionParameterType.BOOLEAN:
+      return z.boolean();
+    case InputDefinitionParameterType.LIST:
+      return z.array(
+        z.union([stringSchema, structSchema, z.number(), z.number().int(), z.boolean()]),
+      );
```

Comment:
i think we dont need to complicate this and as long as it is an array we are fine

```suggestion
      return z.array(z.any());
```